client-examples/BrutalmazeClient.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Net;
+using System.Net.Sockets;
+using System.Diagnostics;
+
+namespace BrutalmazeClient
+{
+	class Program
+	{
+		private static Socket ConnectSocket(string host, int port)
+		{
+			IPHostEntry ipHostInfo = Dns.GetHostEntry(host);
+			IPAddress ipAdress = ipHostInfo.AddressList[0];
+			IPEndPoint remoteEP = new IPEndPoint(ipAdress, port);
+			Socket tmp = new Socket(ipAdress.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+			tmp.Connect(remoteEP);
+			return tmp;
+		}
+
+		static void Main(string[] args)
+		{
+			Process.Start("CMD.exe", "/C brutalmaze --server");
+			System.Threading.Thread.Sleep(1000);
+			const string host = "localhost";
+			const int port = 8089;
+			Socket client_socket = ConnectSocket(host, port);
+			Random rnd = new Random();
+
+			byte[] buff = new byte[1000];
+			byte[] query;
+			string[] matrix = new string[100];
+			const int MAGIC = 42; // For escape
+			int prevX = 1234, prevY = 5678;
+			int dir = 0, deg = 0, atk = 1;
+			int needEsc = 0;
+
+			while (42 < 420)
+			{
+				int recv, sent;
+				try
+				{ 
+					recv = client_socket.Receive(buff, 7, 0);
+				} catch (SocketException e)
+				{
```

Comment:
Ugh, this is ugly. IMHO style for code blocks should be symmetric, i.e. either

```csharp
try {
	/* foo */
} catch (bar) {
	/* baz */
}
```

(Linux style) or the, well, other style (I will try not to mention GNU style here because I don't think you would prefer 2-space indentation):

```csharp
try
{
	/* foo */
}
catch (bar)
{
	/* baz */
}
```

Also the compiler says

```csharp
BrutalmazeClient.cs(43,9): warning CS0219: The variable `recv' is assigned but its value is never u
sed
BrutalmazeClient.cs(43,15): warning CS0219: The variable `sent' is assigned but its value is never 
used
BrutalmazeClient.cs(47,30): warning CS0168: The variable `e' is declared but never used
```

Is it wise to redeclare these var every ~16ms loop? I think you should trynna move them outa scope. And there is a trailing space on L45.